extensions/blocks/opentable/opentable.php

```diff
@@ -72,10 +75,53 @@ function load_assets( $attributes ) {
 		$classes
 	);
 	$content = '<div class="' . esc_attr( $classes ) . '">';
-	// The OpenTable script uses multiple `rid` paramters,
-	// so we can't use WordPress to output it, as WordPress attempts to validate it and removes them.
-	// phpcs:ignore WordPress.WP.EnqueuedResources.NonEnqueuedScript
-	$content .= '<script type="text/javascript" src="' . esc_url( build_embed_url( $attributes ) ) . '"></script>';
+
+	$script_url = build_embed_url( $attributes );
+
+	if ( Jetpack_AMP_Support::is_amp_request() ) {
+		// Extract params from URL since it had jetpack_opentable_block_url filters applied.
+		$url_query = \wp_parse_url( $script_url, PHP_URL_QUERY ) . '&overlay=false&disablega=false';
+
+		$src = "https://www.opentable.com/widget/reservation/canvas?$url_query";
+
+		// @todo This is temporary workaround! The height should not be needed, and otherwise it should just have layout=fill. To replace once https://github.com/ampproject/amphtml/issues/29398 fixed.
+		$height = $is_multi ? 361 : 301;
+		$params = array();
+		wp_parse_str( $url_query, $params );
+		if ( 'tall' === $params['theme'] ) {
+			$height = $is_multi ? 550 : 490;
+		} elseif ( 'wide' === $params['theme'] ) {
+			$height = 150;
+		} elseif ( 'button' === $params['theme'] ) {
+			$height = 113;
+		}
+		$layout_attrs = sprintf( 'layout="fixed-height" height="%d"', $height );
```

Comment:
Once https://github.com/ampproject/amphtml/issues/29398 is fixed, this can be just:

```suggestion
		$layout_attrs = 'layout="fill"';
```

And then no need for the `$is_multi` variable.

Reply:
Looks like that bug got fixed and merged in https://github.com/ampproject/amphtml/pull/30167 - does that mean we are good to change this implementation now?

Reply:
It's not yet in production. See the link to the playground test in https://github.com/ampproject/amphtml/issues/29398#issuecomment-687720364.

Reply:
The fix is currently in the AMP beta channel. I've pushed up the fix to use `layout=fill`. I've updated the PR description to indicate how the AMP beta channel can be used to test.